bentoml/saved_bundle/loader.py

```diff
@@ -32,6 +32,130 @@
 logger = logging.getLogger(__name__)
 
 
+def resolve_remote_bundle(method):
+    def wrapper(loader: 'SavedBundleLoader', *args):
+        if _is_remote_path(loader.bundle_path):
+            with _resolve_remote_bundle_path(loader.bundle_path) as local_bundle_path:
+                new_loader = SavedBundleLoader(
+                    local_bundle_path, loader.api_config_override
+                )
+                return method(new_loader, *args)
+
+        return method(loader, *args)
+
+    return wrapper
+
+
+class SavedBundleLoader:
```

Comment:
I really like the refactoring of this module here, it makes things a lot cleaner.

Although I feel adding `api_config_override` to the loader or the BentoService class is probably not the best way to implement this feature here.  It probably makes more sense to just make it an additional parameter of the API server class

Reply:
I'd suggest make `mb_max_latency` and `mb_max_batch_size`  two optional parameters of MarshalService and GunicornMarshalServer, and pass them down from CLI args here: https://github.com/bentoml/BentoML/blob/master/bentoml/server/__init__.py

cc @bojiang thoughts on this?

Reply:
Ah, ok. I thought we want to keep BentoService API metada compatible with what is in MasrshalService. I will do it then how you suggest as it is much simpler. I will also submit the refactoring of the bundle loader in a separate PR.

Reply:
Sounds great! Thanks @jsemric